### PR TITLE
Add fixed_fee + allowlisted contracts

### PR DESCRIPTION
**Pull request by Martoon-00**
Add the missing contract versions.

### CODE REVIEW

**Review by jacobarluck — APPROVED** [review]

**Review by pasqu4le — APPROVED** [review]
Only one nitpick comment.

I have another about docs tho, the `english_auction` `README` has a generic definition that still works, but the `fixed_price_sale` one specifically mentions the different `Modification`s on the contracts and what happens there. Perhaps the latter should now be updated?

Everything else LGTM

File: packages/minter-contracts/ligo/src/fixed_price_sale/fixed_price_sale_market_tez_fixed_fee_allowlisted.mligo [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  { market_storage : storage
+  ; allowlist : allowlist
+  }
+
+let check_allowed_param (param, allowlist : market_entry_points * allowlist) : unit =
+  match param with
+  | Sell p -> begin
+      check_address_allowed (p.sale_token_param_tez.token_for_sale_address, allowlist, "SALE_ADDRESS_NOT_ALLOWED");
+      unit
+      end
```
[the remark this answers is not in the document]
Comment (Martoon-00) [reply packages/minter-contracts/ligo/src/fixed_price_sale/fixed_price_sale_market_tez_fixed_fee_allowlisted.mligo]: Sounds fair, thanks. I updated.

**Review by EGuenz — APPROVED** [review]
This change looks good to me but I have a question about allowlist in general. 

Why is the data structure used for allowlist `Big_map`? I would think that `set` would do the trick. `Big_map` calls are expensive and I think there will generally be only a few tokens whitelisted so contract storage won't fill up with allowlisted tokens. What do you think?

**Comment by Martoon-00** [comment]
> This change looks good to me but I have a question about allowlist in general.
> 
> Why is the data structure used for allowlist `Big_map`? I would think that `set` would do the trick. `Big_map` calls are expensive and I think there will generally be only a few tokens whitelisted so contract storage won't fill up with allowlisted tokens. What do you think?

Hm, fair, my bad. Looks like `big_map` access has constant, but relatively high-cost access similarly to other operations with it. Experiments show that `big_map` starts winning `set` only with more than 5 entries in the allowlist, and `set` gets ~`13ꜩ` more expensive per entry in the allowlist.

@jacobarluck what do you think? Can we update this at the current point?

**Comment by Martoon-00** [comment]
Oh, actually I remembered the reason for `big_map`: if we call other entrypoints that don't need access to the allowlist, they are cheaper with `big_map` than with `set` (having `big_map` with any content is even cheaper than having a set with only one address, and that makes sense since addresses in `big_map` get deserialized on demand, unlike in `set`).

So for the marketplace contract, the costs get compensated (`Sell` is more expensive, but `Buy`/`Cancel` are cheaper), while the auction contract that allows multiple bids so it gets cheaper with `big_map`.

**Pull request merged**